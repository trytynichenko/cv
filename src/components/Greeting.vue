<template>
  <div class="greeting block">
    <div class="title">
      <h1>Hey there,<img src="../assets/images/v.png"/></h1>
      <h2>
        My name is <strong>{{ name }}</strong> and I am a Front-end (React) software engineer.
        I have {{ experience }}+ years of experience in e-commerce building beautiful interfaces
        and providing responsive front-end development and WordPress websites for the clients
        from all over the world. As a team lead, I take the responsibility for providing guidance,
        instruction, direction, and leadership to the team for the purpose of reaching the business
        goals and achieving successful results.
      </h2>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Greeting',
  data: () => ({
    from: 2010,
  }),
  props: {
    name: String,
  },
  computed: {
    experience: data => new Date().getFullYear() - data.from,
  },
};
</script>

<style scoped lang="scss">
  @import '../assets/styles/components/greeting';
</style>
