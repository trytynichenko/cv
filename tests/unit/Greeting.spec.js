import { shallow } from '@vue/test-utils';
import Greeting from '@/components/Greeting.vue';

const props = {
  name: 'Bob',
};

const wrapper = shallow(Greeting, {
  propsData: props,
});

describe('Greeting.vue', () => {
  it('renders props.name when passed', () => {
    expect(wrapper.text()).toMatch(props.name);
  });
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
