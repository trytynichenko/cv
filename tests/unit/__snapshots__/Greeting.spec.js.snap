// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting.vue matches snapshot 1`] = `
<div class="greeting block">
  <div class="title">
    <h1>Hey there,
      <img src="../assets/images/v.png">
    </h1>
    <h2>
      My name is <strong>Bob</strong> and I am a Software Engineer. I have 8+ years of experience in software development. As a team lead, I take the responsibility for providing guidance, instruction, direction, and leadership to the team for the purpose
      of reaching the business goals and achieving successful results.
    </h2>
  </div>
</div>
`;
